src/ImageSharp/Formats/Qoi/QoiDecoderCore.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using SixLabors.ImageSharp.IO;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.Metadata;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Formats.Qoi;
+
+internal class QoiDecoderCore : IImageDecoderInternals
+{
+    /// <summary>
+    ///     The global configuration.
+    /// </summary>
+    private readonly Configuration configuration;
+
+    /// <summary>
+    ///     Used the manage memory allocations.
+    /// </summary>
+    private readonly MemoryAllocator memoryAllocator;
+
+    /// <summary>
+    ///     Gets or sets a value indicating whether the metadata should be ignored when the image is being decoded.
+    /// </summary>
+    private readonly bool skipMetadata;
+
+    /// <summary>
+    ///     The QOI header.
+    /// </summary>
+    private QoiHeader header;
+
+    public QoiDecoderCore(DecoderOptions options)
+    {
+        this.Options = options;
+        this.configuration = options.Configuration;
+        this.skipMetadata = options.SkipMetadata;
+        this.memoryAllocator = this.configuration.MemoryAllocator;
+    }
+
+    public DecoderOptions Options { get; }
+
+    public Size Dimensions { get; }
+
+    public Image<TPixel> Decode<TPixel>(BufferedReadStream stream, CancellationToken cancellationToken)
+        where TPixel : unmanaged, IPixel<TPixel> => throw new NotImplementedException();
+
+    public ImageInfo Identify(BufferedReadStream stream, CancellationToken cancellationToken)
+    {
+        ImageMetadata metadata = new();
```

Comment:
You'd need to implement a `QoiMetadata` and expose stuff like `QoiChannels` and `QoiColorSpace` there.